apollo-federation/src/query_graph/path_tree.rs

```diff
@@ -334,6 +368,16 @@ where
                         (Some(cond_a), Some(cond_b)) => cond_a.equals_same_root(cond_b),
                         _ => false,
                     }
+                    && match (&a.context_to_selection, &b.context_to_selection) {
+                        (None, None) => true,
+                        (Some(set_a), Some(set_b)) => set_a == set_b,
+                        _ => false,
+                    }
```

Comment:
This should just be `a.context_to_selection == b.context_to_selection`, no?
```suggestion
                    && a.context_to_selection** == b.context_to_selection
```

Reply:
good call